lib/adapters/http_server.js

```diff
@@ -89,9 +88,44 @@ var Read = source.extend({
             port: this.port,
             path: '/'
         });
-    },
+    }
 
-    _handle_listen_error: function(res, err) {
+    read(from, to, limit, state) {
+        this.logger.debug('in read --', this.points.length, 'buffered points');
+        if (this.points.length > 0) {
+            return this.readResult(limit);
+        } else {
+            return new Promise((resolve, reject) => {
+                this.pendingRead = resolve;
+                this.pendingReadLimit = limit;
+            })
+            .timeout(this.readEvery)
+            .catch(Promise.TimeoutError, (err) => {
+                this.logger.debug('read: no points arrived in', this.readEvery, 'ms');
+                return {};
+            });
+        }
+    }
+
+    readResult(limit) {
+        var ret = {};
+        if (this.points.length <= limit) {
```

Comment:
Computers are fast! It wouldn't reallocate all the objects so we're talking about like 10 microseconds here.